tests/integration/test_app.py

```diff
@@ -2,41 +2,63 @@
 #  See LICENSE file for licensing details.
 
 """Integration tests for the flask app."""
+import hashlib
+import hmac
+import json
 import os
 import random
+import secrets
 
 # security considerations for subprocess are not relevant in this module
 import subprocess  # nosec
 from pathlib import Path
 from time import sleep
-from typing import Iterator
+from typing import Iterator, Optional
 
 import pytest
 import requests
 
+from src.app import WEBHOOK_SIGNATURE_HEADER
+
 BIND_HOST = "localhost"
 BIND_PORT = 5000
 BASE_URL = f"http://{BIND_HOST}:{BIND_PORT}"
 
 
-@pytest.fixture(name="webhook_logs_dir", scope="module")
-def webhook_logs_dir_fixture(tmp_path_factory: pytest.TempPathFactory) -> Path:
+@pytest.fixture(name="webhook_logs_dir")
+def webhook_logs_dir_fixture(tmp_path: Path) -> Path:
     """Create a dir for the webhook logs."""
-    logs_dir = tmp_path_factory.mktemp("webhook_logs")
+    logs_dir = tmp_path / "webhook_logs"
+    logs_dir.mkdir()
     return logs_dir
 
 
-@pytest.fixture(name="process_count", scope="module")
+@pytest.fixture(name="process_count")
 def process_count_fixture() -> int:
     """Return the number of processes."""
     # We do not use randint for cryptographic purposes.
     return random.randint(2, 5)  # nosec
 
 
-@pytest.fixture(name="app", scope="module", autouse=True)
-def app_fixture(webhook_logs_dir: Path, process_count: int) -> Iterator[None]:
+@pytest.fixture(
+    name="webhook_secret",
+    params=[pytest.param(True, id="with_secret"), pytest.param(False, id="without_secret")],
+)
+def webhook_secret_fixture(request) -> Optional[str]:
+    """Return a webhook secret or None."""
+    return secrets.token_hex(16) if request.param else None
+
+
+@pytest.fixture(name="app", autouse=True)
+def app_fixture(
+    webhook_logs_dir: Path, process_count: int, webhook_secret: Optional[str]
+) -> Iterator[None]:
     """Setup and run the flask app."""
     os.environ["WEBHOOK_LOGS_DIR"] = str(webhook_logs_dir)
+    if webhook_secret:
+        os.environ["WEBHOOK_SECRET"] = webhook_secret
+    else:
+        os.environ.pop("WEBHOOK_SECRET", None)
```

Comment:
I think pytest's monkeypatch setenv should work here: https://docs.pytest.org/en/latest/how-to/monkeypatch.html